thcrap_proton

```diff
@@ -90,10 +121,20 @@ echo "$COMMAND"
 echo "---------------------------------------------------------"
 
 sh -c "$COMMAND" &
+if [[ "$USE_THPRAC" == 1 && -z $(echo "$COMMAND" | grep -E 'custom\.exe$') ]]; then
+    # wait 5s and launch thprac so it could auto attach to the game
+    sleep 5
+    PROTON_CMD=$( echo "$COMMAND" | sed -n "s/^.*-- \(.*\) waitforexitandrun.*$/\\1/p" )
+    RUN_THPRAC="$PROTON_CMD run '$THPRAC_FOLDER/thprac.exe'"
+    echo "-----------------------THPRAC COMMAND--------------------"
+    echo "$RUN_THPRAC"
+    echo "---------------------------------------------------------"
+    sh -c "$RUN_THPRAC" &
+fi
 
 # Get PID while ensuring only the actual Touhou executable is chosen
 while [ -z "$TH_PID" ]; do # Loop until a matching process is found
-    TH_PID=$(grep -Er "$TH_REGEX" /proc/*/stat | sed "s/.*:\([0-9]\+\).*/\\1/")
+    TH_PID=$(grep -Er "($TH_REGEX\.exe|custom\.exe)" /proc/*/stat | sed "s/.*:\([0-9]\+\).*/\\1/")
```

Comment:
This will not work for EoSD because the executable name in `/proc/[pid]/stat` is limited to 15 bytes, and `東方紅魔郷.exe` is 19 bytes, so it gets truncated to just `東方紅魔郷`.

Reply:
> Okay, I didn't know that. Does `"($TH_REGEX|custom\.exe)"` works?

`"($TH_REGEX|custom\.exe)"` should work fine.